### PR TITLE
Generate KerasHub snippets based on tasks from `metadata.json`

### PR DESCRIPTION
**Pull request by Wauplin**
This PR updates the `keras-hub` snippets based on the new `metadata.json > tasks` field. This field is now uploaded for all KerasHub models (see https://github.com/keras-team/keras-hub/pull/1997) and contains the list of tasks compatible for a given model. This allows us to generate multiple snippets when relevant. For instance, [keras/stable_diffusion_3.5_large_turbo](https://huggingface.co/keras/stable_diffusion_3.5_large_turbo/blob/main/metadata.json) is compatible with `ImageToImage`, `Inpaint` and `TextToImage` tasks.

For this PR to work, we'll need to parse the metadata.json file server-side. This is done in https://github.com/huggingface-internal/moon-landing/pull/11693 (private PR). We can merge these 2 PRs independently.

cc @martin-gorner @mattdangerw @SamanehSaadat who coordinated this 

---

**Note:** I also removed the legacy `keras-nlp` library (only [18 remaining models](https://huggingface.co/models?library=keras-nlp)).

### CODE REVIEW

**Review by julien-c — APPROVED** [review]
let's wait for some reviews from the Keras side, but looks great to me!! 💥

Thanks for pushing this accross the finish line, we will have great support for Keras on the huggingface Hub. cc @fchollet for visibility too

File: packages/tasks/src/model-libraries-snippets.ts [hunk 1]
```diff
@@ -394,32 +394,91 @@ model = keras.saving.load_model("hf://${model.id}")
...
+
+# Classify image
+image_classifier.predict(keras.ops.ones((1, 64, 64, 3)))
+`;
+
+const _keras_hub_tasks_with_example = {
+	CausalLM: _keras_hub_causal_lm,
+	TextToImage: _keras_hub_text_to_image,
+	TextClassifier: _keras_hub_text_classifier,
+	ImageClassifier: _keras_hub_image_classifier,
```
Comment (divyashreepathihalli) [inline packages/tasks/src/model-libraries-snippets.ts:442]: we have a few more for ImageSegmenter, ImagetoImage

Comment (Wauplin) [reply packages/tasks/src/model-libraries-snippets.ts:442]: That'd be great! Would be nice if we can settle this PR with the existing snippets first and once we agreed on the format / what we put in the snippets then open a new PR with remaining tasks

File: packages/tasks/src/model-libraries-snippets.ts [hunk 2]
```diff
@@ -394,32 +394,91 @@ model = keras.saving.load_model("hf://${model.id}")
...
-];
+# Classify text
+text_classifier.predict("Keras is a multi-backend ML framework.")
+`;
+
+const _keras_hub_image_classifier = (modelId: string): string => `
+import keras_hub
+
+# Load ImageClassifier model
+text_classifier = keras_hub.models.ImageClassifier.from_preset(${modelId})
```
Comment (mattdangerw) [inline packages/tasks/src/model-libraries-snippets.ts:432]: Should be image_classifier I think.

File: packages/tasks/src/model-libraries-snippets.ts [hunk 3]
```diff
@@ -394,32 +394,91 @@ model = keras.saving.load_model("hf://${model.id}")
...
+# Generate images with a TextToImage model.
+text_to_image.generate("Astronaut in a jungle")
+`;
 
+const _keras_hub_text_classifier = (modelId: string): string => `
 import keras_hub
 
-# Load a task-specific model (*replace CausalLM with your task*)
-model = keras_hub.models.CausalLM.from_preset("hf://${model.id}", dtype="bfloat16")
+# Load TextClassifier model
```
Comment (mattdangerw) [inline packages/tasks/src/model-libraries-snippets.ts:421]: Because our most popular classifiers won't have pretrained heads (thinking of bert likes). I think this makes sense to show with some toy fine-tuning. E.g.

```python
import keras_hub
import keras

# Load TextClassifier model
text_classifier = keras_hub.models.TextClassifier.from_preset(
    ${modelId},
    num_classes=2,
)
# Fine-tune
text_classifier.fit(x=["Thilling adventure!", "Total snoozefest."], y=[1, 0])
# Classify text
text_classifier.predict(["Not my cup of tea."])
```

File: packages/tasks/src/model-libraries-snippets.ts [hunk 4]
```diff
@@ -394,32 +394,91 @@ model = keras.saving.load_model("hf://${model.id}")
...
-# Possible tasks are CausalLM, TextToImage, ImageClassifier, ...
-# full list here: https://keras.io/api/keras_hub/models/#api-documentation
-`,
-];
+# Classify text
+text_classifier.predict("Keras is a multi-backend ML framework.")
+`;
+
+const _keras_hub_image_classifier = (modelId: string): string => `
+import keras_hub
```
Comment (mattdangerw) [inline packages/tasks/src/model-libraries-snippets.ts:429]: Same here. Something like

```python
import keras_hub
import keras

# Load ImageClassifier model
image_classifier = keras_hub.models.ImageClassifier.from_preset(
    "resnet_50_imagenet",
    num_classes=2,
)
# Fine-tune
image_classifier.fit(
    x=keras.random.randint((32, 64, 64, 3), 0, 256),
    y=keras.random.randint((32, 1), 0, 2),
)
# Classify image
image_classifier.predict(keras.random.randint((1, 64, 64, 3), 0, 256))
```

**Review by mattdangerw — APPROVED** [review]
Looks great! Mostly comments on the classifiers. I think showing them with a fit call will make the snippets a little more general.

**Review by divyashreepathihalli — APPROVED** [review]
Sounds good! LGTM!

File: packages/tasks/src/model-libraries-snippets.ts (hunk 3 above)
Comment (Wauplin) [reply packages/tasks/src/model-libraries-snippets.ts:421]: Is there a reason why we need to provide `num_classes=2` ? Shouldn't this depend on the model we chose ?

I feel that we should aim at showcasing how to use the model on the specific task it has been trained on (i.e. the simplest use case). If we add a `num_classes=2` + a `.fit(...)` step, then a normal user won't be able to just use the model. They would need to know they have to remove the `num_classes` parameter.

---

@mattdangerw  If we want to show how to finetune + other use cases, what do you think about linking to the base class documentation page like this?

```py
# Check out https://keras.io/keras_hub/api/base_classes/text_classifier/ for more examples.
import keras_hub

# Load TextClassifier model
text_classifier = keras_hub.models.TextClassifier.from_preset(${modelId})

# Classify text
text_classifier.predict("Keras is a multi-backend ML framework.")
```

Comment (julien-c) [reply packages/tasks/src/model-libraries-snippets.ts:421]: no super strong opinion but in our experience it's good to keep snippets as minimal as possible (and focus on predict rather than fit)

Your call in the end though!

Comment (mattdangerw) [reply packages/tasks/src/model-libraries-snippets.ts:421]: Thanks!

The issue is really with the bert-likes (but those are fairly prominent in quick starts, etc). Since these have no built in head, if you try to create those without specifying `num_classes` you actually get an error. Something like...

```python
>>> keras_hub.models.TextClassifier.from_preset("roberta_base_en")
Error you must supply num_classes to the classifier
```

I know the transformer answer here is to default these as "fill mask" models. And we could consider something like that but I think it's probably better done as a follow up. Almost everyone will be using these models for classification, and the code I'm showing above will work generally for classifiers (unlike the code in the PR currently).

Let me know if that makes sense! We could also ditch the `fit()` line, but we'd be showing random predictions because the heads won't be trained.

Comment (Wauplin) [reply packages/tasks/src/model-libraries-snippets.ts:421]: Makes perfect sense! Then we'll go with what you've suggested, e.g. num_classes + .fit example. Thanks for the explanation :)

Comment (Wauplin) [reply packages/tasks/src/model-libraries-snippets.ts:421]: updated in https://github.com/huggingface/huggingface.js/pull/1118/commits/e608ac0b8a7927271a7f176ad46a0bd01ec18ca0

File: packages/tasks/src/model-libraries-snippets.ts (hunk 4 above)
Comment (Wauplin) [reply packages/tasks/src/model-libraries-snippets.ts:433]: https://github.com/huggingface/huggingface.js/pull/1118/commits/e608ac0b8a7927271a7f176ad46a0bd01ec18ca0

**Comment by Wauplin** [comment]
(failing test is unrelated)
